src/mono/mono/component/debugger-agent.c

```diff
@@ -1636,6 +1636,10 @@ mono_init_debugger_agent_for_wasm (int log_level_parm, MonoProfilerHandle *prof)
 
 	agent_config.enabled = TRUE;
 
+	start_debugger_thread_func = start_debugger_thread;
+	suspend_vm_func = suspend_vm;
```

Comment:
Not sure about this one: it's main side-effect looks like it will enable single-stepping in the current (only) thread. I see we have `mono_wasm_single_step_hit` as a single step trampoline for wasm, so that seems ok.  But I'm not sure if we expect to be in single-step mode on wasm after various events.  /cc @thaystg

Reply:
It is used here
https://github.com/dotnet/runtime/blob/34544b9416eb46c1397bc76dde257b4f5209d1a1/src/mono/mono/component/debugger-agent.c#L3768
I'm not 100% sure but I assume that it's because on the client side, I enable type-loading events. The common implementation is then offering a suspend policy different from NONE.